feat: add `ctx.renderNotFound()` to explicitly render 404 template (#481)
denoland/fresh: docs/concepts/error-pages.md

matching route exists, and when a middleware, route handler, or page component
throws an error respectively.

> ### 404

Reviewer: ```suggestion
### 404: Not Found
```

}
```

> #### Manually render 404

Reviewer: ```suggestion
#### Manually render 404 pages
```

> The `_404.tsx` file will be invoke when not routes matches the URL. In some
> cases, you would want to manually trigger the rendering of the 404 page with
> `renderNotFound`.

Reviewer: ```suggestion
The `_404.tsx` file will be invoked automatically when no route matches the URL. In some
cases, one needs to manually trigger the rendering of the 404 page, for example when the route did match, but the requested resource does not exist. This can be achieved with
`ctx.renderNotFound`.
```

```tsx
/** @jsx h */
import { h } from "preact";
import { Handlers, PageProps } from "$fresh/server.ts";

export handler: Handlers = {
  async GET(req, ctx) {
    const blogpost = await fetchBlogpost(ctx.params.slug)
    if (!blogpost) {
      return ctx.renderNotFound();
    }
    return ctx.render({ blogpost })
  }
}

export default function BlogpostPage({ data }) {
  return (
    <article>
      <h1>{data.blogpost.title}</h1>
      {/* rest of your page */}
    </article>
  );
}
```

> ### 500

Reviewer: ```suggestion
### 500: Internal Server Error
```
